src/Common/Core/Impl/Extensions/TaskExtensions.cs

```diff
@@ -122,8 +122,16 @@ public static class TaskExtensions {
         /// within the timeout specified.
         /// </summary>
         public static T WaitTimeout<T>(this Task<T> task, int msTimeout) {
-            task.Wait(msTimeout);
-            return task.IsCompleted ? task.Result : default(T);
+            bool result = false;
+            try {
+                task.Wait(msTimeout);
+                result = task.IsCompleted;
+            } catch(AggregateException) { }
```

Comment:
`Wait` does not throw anything. `Result` will throw if task completed with an exception.